frontend/src/metabase/query_builder/components/expressions/CombineColumns/CombineColumns.tsx

```diff
@@ -131,33 +157,33 @@ export function CombineColumns({ query, stageIndex, onSubmit, width }: Props) {
     isNotNull(column),
   );
 
-  const example = useMemo(
-    () => getExample(state.columnsAndSeparators),
-    [state.columnsAndSeparators],
-  );
-
   return (
     <form onSubmit={handleSubmit}>
       <Box maw="100vw" w={width} p="lg" pt={0}>
         <Stack spacing="lg" mt="lg">
           <Stack spacing="md">
             <Box>
               <Stack spacing="md">
-                {columnsAndSeparators.map(({ column, separator }, index) => (
-                  <ColumnAndSeparatorRow
-                    key={index}
-                    query={query}
-                    stageIndex={stageIndex}
-                    index={index}
-                    columns={expressionableColumns}
-                    column={column}
-                    separator={separator}
-                    showSeparator={!isUsingDefaultSeparator && index !== 0}
-                    showRemove={columnsAndSeparators.length >= 3}
-                    onChange={handleRowChange}
-                    onRemove={handleRowRemove}
-                  />
-                ))}
+                {columnsAndSeparators.map(
+                  (item, index) =>
+                    // Do not allow editing the first column when it is passed from
+                    // the props.
+                    (!column || index > 0) && (
```

Comment:
AFAICT React does not need keys on literal array elements (they kind of act as their own key). 

On top of that `false` does not get rendered, so it does not need an identity (since it cannot be added, removed or moved).

If you want to test this, try rendering:

```tsx
<div>{[1, false, null, "foo", true, <span key="key" />]}</div>
```

This does not log a warning about missing keys, while:
```tsx
<div>{[1, false, null, "foo", true, <span />]}</div>
```
does.